mesonbuild/scripts/env2mfile.py

```diff
@@ -175,6 +184,8 @@ def detect_cross_debianlike(options: T.Any) -> MachineInfo:
     infos.binaries['strip'] = locate_path("%s-strip" % host_arch)
     infos.binaries['objcopy'] = locate_path("%s-objcopy" % host_arch)
     infos.binaries['ld'] = locate_path("%s-ld" % host_arch)
+    if cmake:
+        infos.binaries['cmake'] = cmake
```

Comment:
Setting a cmake binary without also setting the variables is actively harmful, because it'll cause cmake to search for native things. Please either do the complete change or defer the cmake binary.